Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -2299,13 +2299,17 @@ bool KeyframeEffect::ticksContinouslyWhileActive() const
 
 Seconds KeyframeEffect::timeToNextTick(const BasicEffectTiming& timing) const
 {
-    if (timing.phase == AnimationEffectPhase::Active) {
-        // CSS Animations need to trigger "animationiteration" events even if there is no need to
-        // update styles while animating, so if we're dealing with one we must wait until the next iteration.
-        if (!ticksContinouslyWhileActive() && is<CSSAnimation>(animation())) {
-            if (auto iterationProgress = getComputedTiming().simpleIterationProgress)
-                return iterationDuration() * (1 - *iterationProgress);
-        }
+    // CSS Animations need to trigger "animationiteration" events even if there is no need to
+    // update styles while animating, so if we're dealing with one we must wait until the next iteration.
+    // We only do this in case any CSS Animation event was registered since, in the general case, there's
+    // a good chance that no such event listeners were registered and we can avoid some unnecessary
+    // animation resolution scheduling.
```

Comment:
Pptimally we would not even enter style update code if event generation is the only reason we are ticking.

Reply:
Agreed, I filed [bug 265940](https://bugs.webkit.org/show_bug.cgi?id=265940).